### PR TITLE
Upgrade from Chromium 106 to Chromium 107

### PR DESCRIPTION
**Pull request by cdesouza-chromium**
Resolves https://github.com/brave/brave-browser/issues/25658

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: ios/browser/api/bookmarks/bookmark_model_listener_ios.h [hunk 1]
```diff
@@ -38,7 +38,8 @@ class BookmarkModelListener : public bookmarks::BookmarkModelObserver {
                          size_t new_index) override;
   void BookmarkNodeAdded(bookmarks::BookmarkModel* model,
                          const bookmarks::BookmarkNode* parent,
-                         size_t index) override;
+                         size_t index,
+                         bool added_by_user) override;
```
Comment (kylehickinson) [inline ios/browser/api/bookmarks/bookmark_model_listener_ios.h:42]: cc @soner-yuksel in case there's something we need to do here

File: ios/browser/api/bookmarks/brave_bookmarks_api.mm [hunk 2]
```diff
@@ -159,7 +159,8 @@ - (bool)isPermanentNode {
 - (void)setTitle:(NSString*)title {
   DCHECK(node_);
   DCHECK(model_);
-  model_->SetTitle(node_, base::SysNSStringToUTF16(title));
+  model_->SetTitle(node_, base::SysNSStringToUTF16(title),
+                   bookmarks::metrics::BookmarkEditSource::kUser);
```
Comment (kylehickinson) [inline ios/browser/api/bookmarks/brave_bookmarks_api.mm:163]: cc @soner-yuksel

I think for the most part setting the title/url is going to be done by the user so probably nothing to change here. The other options are `kExtension` and `kOther`

File: vendor/bat-native-ads/src/bat/ads/ad_content_value_util.cc [hunk 3]
```diff
@@ -68,88 +68,103 @@ base::Value::Dict AdContentToValue(const AdContentInfo& ad_content) {
...
 
-  if (const auto* value = root.FindString(kAdType)) {
-    ad_content.type = AdType(*value);
-  } else if (const auto* value = root.FindString(kLegacyAdType)) {
-    ad_content.type = AdType(*value);
+  if (const auto* ad_type_value = root.FindString(kAdType)) {
+    ad_content.type = AdType(*ad_type_value);
+  } else if (const auto* legacy_ad_type_value =
+                 root.FindString(kLegacyAdType)) {
+    ad_content.type = AdType(*legacy_ad_type_value);
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/ad_content_value_util.cc:75]: Are these changes needed?

File: vendor/bat-native-ads/src/bat/ads/history_item_value_util.cc [hunk 4]
```diff
@@ -54,21 +54,23 @@ base::Value::List HistoryItemToDetailRowsValue(
 HistoryItemInfo HistoryItemFromValue(const base::Value::Dict& root) {
   HistoryItemInfo history_item;
 
-  if (const auto* value = root.FindString(kCreatedAtKey)) {
+  if (const auto* created_at_value_string = root.FindString(kCreatedAtKey)) {
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/history_item_value_util.cc:57]: Are these changes needed?

File: vendor/bat-native-ads/src/bat/ads/internal/ads/serving/eligible_ads/priority/priority.h [hunk 5]
```diff
@@ -26,9 +26,10 @@ T PrioritizeCreativeAds(const T& creative_ads) {
     return {};
   }
 
-  const std::pair<unsigned int, T> bucket = GetHighestPriorityBucket(buckets);
-  const unsigned int priority = bucket.first;
-  const T prioritized_creative_ads = bucket.second;
+  const std::pair<unsigned int, T> highest_priority_bucket =
+      GetHighestPriorityBucket(buckets);
+  const unsigned int priority = highest_priority_bucket.first;
+  const T prioritized_creative_ads = highest_priority_bucket.second;
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/ads/serving/eligible_ads/priority/priority.h:32]: Is this change needed?

File: vendor/bat-native-ads/src/bat/ads/internal/creatives/inline_content_ads/creative_inline_content_ads_database_table.cc [hunk 6]
```diff
@@ -184,15 +184,19 @@ void CreativeInlineContentAds::Save(
   for (const auto& batch : batches) {
     InsertOrUpdate(transaction.get(), batch);
 
-    const CreativeAdList creative_ads(batch.cbegin(), batch.cend());
-    campaigns_database_table_->InsertOrUpdate(transaction.get(), creative_ads);
+    const CreativeAdList creative_ads_list(batch.cbegin(), batch.cend());
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/creatives/inline_content_ads/creative_inline_content_ads_database_table.cc:187]: Is this change needed as plural implies this is a list?

File: vendor/bat-native-ads/src/bat/ads/internal/creatives/new_tab_page_ads/creative_new_tab_page_ads_database_table.cc [hunk 7]
```diff
@@ -190,17 +190,21 @@ void CreativeNewTabPageAds::Save(const CreativeNewTabPageAdList& creative_ads,
   for (const auto& batch : batches) {
     InsertOrUpdate(transaction.get(), batch);
 
-    const CreativeAdList creative_ads(batch.cbegin(), batch.cend());
-    campaigns_database_table_->InsertOrUpdate(transaction.get(), creative_ads);
+    const CreativeAdList creative_ads_list(batch.cbegin(), batch.cend());
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/creatives/new_tab_page_ads/creative_new_tab_page_ads_database_table.cc:193]: See previous comment, thanks

File: vendor/bat-native-ads/src/bat/ads/ad_content_value_util.cc (hunk 3 above)
Comment (emerick) [reply vendor/bat-native-ads/src/bat/ads/ad_content_value_util.cc:75]: This bump included clang's `-Wshadow` in the overall build process, which disallows shadowed variables. Open to other approaches here, but basically the name does have to change.

File: vendor/bat-native-ads/src/bat/ads/internal/creatives/notification_ads/creative_notification_ads_database_table.cc [hunk 8]
```diff
@@ -176,15 +176,19 @@ void CreativeNotificationAds::Save(
   for (const auto& batch : batches) {
     InsertOrUpdate(transaction.get(), batch);
 
-    const CreativeAdList creative_ads(batch.cbegin(), batch.cend());
-    campaigns_database_table_->InsertOrUpdate(transaction.get(), creative_ads);
+    const CreativeAdList creative_ads_list(batch.cbegin(), batch.cend());
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/creatives/notification_ads/creative_notification_ads_database_table.cc:179]: See previous comment which also applies to other database_table changes. Thanks

File: vendor/bat-native-ads/src/bat/ads/internal/deprecated/client/client_info.cc [hunk 9]
```diff
@@ -64,8 +64,8 @@ base::Value::Dict ClientInfo::ToValue() const {
   base::Value::Dict advertisers;
   for (const auto& [key, value] : seen_advertisers) {
     base::Value::Dict advertiser;
-    for (const auto& [key, value] : value) {
-      advertiser.Set(key, value);
+    for (const auto& [ad_key, ad_value] : value) {
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/deprecated/client/client_info.cc:67]: do we need the verbosity in this file for key/value? (There is another open issue to cleanup structured bindings.

File: vendor/bat-native-ads/src/bat/ads/internal/deprecated/confirmations/confirmation_state_manager.cc [hunk 10]
```diff
@@ -377,16 +377,17 @@ bool ConfirmationStateManager::GetFailedConfirmationsFromDictionary(
   ConfirmationList new_failed_confirmations;
 
   for (const auto& item : *failed_confirmations) {
-    const base::Value::Dict* dict = item.GetIfDict();
-    if (!dict) {
+    const base::Value::Dict* failed_confirmation_dict = item.GetIfDict();
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/deprecated/confirmations/confirmation_state_manager.cc:380]: same, do we need the verbosity? Thanks

File: vendor/bat-native-ads/src/bat/ads/internal/history/category_content_value_util.cc [hunk 11]
```diff
@@ -31,16 +31,17 @@ base::Value::Dict CategoryContentToValue(
 CategoryContentInfo CategoryContentFromValue(const base::Value::Dict& root) {
   CategoryContentInfo category_content;
 
-  if (const auto* value = root.FindString(kCategoryKey)) {
-    category_content.category = *value;
+  if (const auto* category_key_value = root.FindString(kCategoryKey)) {
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/history/category_content_value_util.cc:34]: same, do we need the verbosity? Thanks

File: vendor/bat-native-ads/src/bat/ads/internal/ml/pipeline/embedding_pipeline_value_util.cc [hunk 12]
```diff
@@ -49,8 +49,8 @@ absl::optional<EmbeddingPipelineInfo> EmbeddingPipelineFromValue(
   }
 
   embedding_pipeline.dimension = 1;
-  for (const auto [key, value] : *value) {
-    const auto* list = value.GetIfList();
+  for (const auto [k, v] : *value) {
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/ml/pipeline/embedding_pipeline_value_util.cc:52]: Can we please revert as we should not use abbreviations. Thanks

File: vendor/bat-native-ads/src/bat/ads/new_tab_page_ad_value_util.cc [hunk 13]
```diff
@@ -46,15 +46,15 @@ base::Value::Dict NewTabPageAdToValue(const NewTabPageAdInfo& ad) {
 
   base::Value::List wallpapers;
   for (const NewTabPageAdWallpaperInfo& wallpaper : ad.wallpapers) {
-    base::Value::Dict dict;
-    dict.Set(kImageUrlKey, wallpaper.image_url.spec());
+    base::Value::Dict wallpaper_dict;
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/new_tab_page_ad_value_util.cc:49]: same, do we need the verbosity? Thanks

File: vendor/bat-native-ads/src/bat/ads/ad_content_value_util.cc (hunk 3 above)
Comment (emerick) [reply vendor/bat-native-ads/src/bat/ads/ad_content_value_util.cc:75]: Nevermind, I see the issue. Will address in a fixup.

File: vendor/bat-native-ads/src/bat/ads/internal/ads/serving/eligible_ads/priority/priority.h (hunk 5 above)
Comment (emerick) [reply vendor/bat-native-ads/src/bat/ads/internal/ads/serving/eligible_ads/priority/priority.h:32]: Yes, it was being shadowed by `bucket` below.

File: vendor/bat-native-ads/src/bat/ads/internal/creatives/new_tab_page_ads/creative_new_tab_page_ads_database_table.cc (hunk 7 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/creatives/new_tab_page_ads/creative_new_tab_page_ads_database_table.cc:193]: Ah yes they are shadowed.

File: vendor/bat-native-ads/src/bat/ads/internal/deprecated/client/client_info.cc (hunk 9 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/deprecated/client/client_info.cc:67]: Ah yes they are shadowed.

File: vendor/bat-native-ads/src/bat/ads/internal/deprecated/confirmations/confirmation_state_manager.cc (hunk 10 above)
Comment (emerick) [reply vendor/bat-native-ads/src/bat/ads/internal/deprecated/confirmations/confirmation_state_manager.cc:380]: This shadows the `dict` function parameter, but please double-check me.

File: vendor/bat-native-ads/src/bat/ads/new_tab_page_ad_value_util.cc (hunk 13 above)
Comment (emerick) [reply vendor/bat-native-ads/src/bat/ads/new_tab_page_ad_value_util.cc:49]: This shadows the `dict` local variable above, but please double-check me.

File: vendor/bat-native-ads/src/bat/ads/ad_content_value_util.cc (hunk 3 above)
Comment (emerick) [reply vendor/bat-native-ads/src/bat/ads/ad_content_value_util.cc:75]: Fixed in b4174716b278ed73d7aab2c1ed841d54e25cb51a.

File: vendor/bat-native-ads/src/bat/ads/history_item_value_util.cc (hunk 4 above)
Comment (emerick) [reply vendor/bat-native-ads/src/bat/ads/history_item_value_util.cc]: Fixed in b4174716b278ed73d7aab2c1ed841d54e25cb51a.

File: vendor/bat-native-ads/src/bat/ads/internal/history/category_content_value_util.cc (hunk 11 above)
Comment (emerick) [reply vendor/bat-native-ads/src/bat/ads/internal/history/category_content_value_util.cc]: Fixed in b4174716b278ed73d7aab2c1ed841d54e25cb51a.

File: vendor/bat-native-ads/src/bat/ads/internal/ml/pipeline/embedding_pipeline_value_util.cc (hunk 12 above)
Comment (emerick) [reply vendor/bat-native-ads/src/bat/ads/internal/ml/pipeline/embedding_pipeline_value_util.cc]: We do need to rename this as value shadows a variable, but I renamed to `embedding_key` and `embedding_value` in b4174716b278ed73d7aab2c1ed841d54e25cb51a.

**Review by tmancey — APPROVED** [review]
LGTM++

File: components/brave_wallet/browser/keyring_service.cc [hunk 14]
```diff
@@ -1404,18 +1409,15 @@ std::vector<mojom::AccountInfoPtr> KeyringService::GetAccountInfosForKeyring(
 std::vector<mojom::AccountInfoPtr> KeyringService::GetHardwareAccountsSync(
     const std::string& keyring_id) const {
   std::vector<mojom::AccountInfoPtr> accounts;
-  const base::Value* value =
+  const auto& keyring =
       GetPrefForKeyringUpdate(prefs_, kHardwareAccounts, keyring_id);
```
[the remark this answers is not in the document]
Comment (emerick) [reply components/brave_wallet/browser/keyring_service.cc]: Thanks, fixed here 9cab7930e32d791e2709037659b6b9031f8bbef6.

File: components/brave_wallet/browser/keyring_service.cc [hunk 15]
```diff
@@ -1404,18 +1409,15 @@ std::vector<mojom::AccountInfoPtr> KeyringService::GetAccountInfosForKeyring(
...
-  if (!value) {
-    return {};
-  }
 
-  for (const auto hw_keyring : value->DictItems()) {
-    std::string device_id = hw_keyring.first;
-    const base::Value* account_value = hw_keyring.second.FindKey(kAccountMetas);
+  for (const auto [id, value] : keyring) {
```
[the remark this answers is not in the document]
Comment (emerick) [reply components/brave_wallet/browser/keyring_service.cc]: You can't unfortunately, as it results in the following error:

```../../brave/components/brave_wallet/browser/keyring_service.cc(1418,20): error: loop variable '[id, value]' binds to a temporary value produced by a range of type 'const base::Value::Dict' [-Werror,-Wrange-loop-bind-reference]
  for (const auto& [id, value] : *keyring) {
```

Updated it to use `auto&&` which seems to be the best practice.

Fixed here 9cab7930e32d791e2709037659b6b9031f8bbef6.

File: components/brave_wallet/browser/keyring_service.cc [hunk 16]
```diff
@@ -1482,24 +1476,25 @@ void KeyringService::RemoveHardwareAccount(
   }
 
   auto keyring_id = GetHardwareKeyringId(coin, address);
-  base::Value* hardware_keyrings =
+  base::Value::Dict& hardware_keyrings =
       GetPrefForKeyringUpdate(prefs_, kHardwareAccounts, keyring_id);
-  for (auto devices : hardware_keyrings->DictItems()) {
-    base::Value* account_metas = devices.second.FindKey(kAccountMetas);
+  for (auto [id, device] : hardware_keyrings) {
```
[the remark this answers is not in the document]
Comment (emerick) [reply components/brave_wallet/browser/keyring_service.cc]: `auto&` results in the following error:

```
../../brave/components/brave_wallet/browser/keyring_service.cc(1484,14): error: non-const lvalue reference to type 'pair<...>' cannot bind to a temporary of type 'pair<...>'
  for (auto& [id, device] : hardware_keyrings) {
             ^            ~
1 error generated.
```

Updated it to use `auto&&` which seems to be the best practice.

Fixed here 9cab7930e32d791e2709037659b6b9031f8bbef6.

File: browser/brave_wallet/keyring_service_unittest.cc [hunk 17]
```diff
@@ -2782,35 +2788,34 @@ TEST_F(KeyringServiceUnitTest, SetSelectedAccount) {
...
+    EXPECT_EQ(*account, "t1h4n7rphclbmwyjcp6jrdiwlfcuwbroxy3jvg33q");
+    EXPECT_TRUE(SetSelectedAccount(&service, &observer, account.value(),
+                                   mojom::CoinType::FIL));
+    EXPECT_EQ(account.value(),
               GetFilecoinSelectedAccount(&service, mojom::kFilecoinTestnet));
   }
   // Can set Solana account
   {
     // lazily create keyring when importing SOL account
-    absl::optional<std::string> imported_account = ImportAccount(
```
[the remark this answers is not in the document]
Comment (emerick) [reply browser/brave_wallet/keyring_service_unittest.cc:2802]: Thanks, fixed here d86c37b689e1ee3342939b38e63b6ae1199fb2d2.

**Review by simonhong — APPROVED** [review]
components/brave_wayback_machine ++

**Review by zenparsing — APPROVED** [review]
Rewards 👍

File: ios/browser/api/bookmarks/bookmark_model_listener_ios.h (hunk 1 above)
Comment (soner-yuksel) [reply ios/browser/api/bookmarks/bookmark_model_listener_ios.h:42]: Yeah we do not need / use if bookmark is added by user or not( in this case **not** part is extension or any other source). So I believe we have no need to update anything for  `bookmarkNodeChildrenChanged:`

File: ios/browser/api/bookmarks/brave_bookmarks_api.mm (hunk 2 above)
Comment (soner-yuksel) [reply ios/browser/api/bookmarks/brave_bookmarks_api.mm:163]: The only different option. user can use to set title and url is after exporting but it is still an interaction with user in our case. Still nothing to do here.

File: components/body_sniffer/body_sniffer_url_loader.cc [hunk 18]
```diff
@@ -66,7 +66,8 @@ void BodySnifferURLLoader::OnReceiveEarlyHints(
 
 void BodySnifferURLLoader::OnReceiveResponse(
     network::mojom::URLResponseHeadPtr response_head,
-    mojo::ScopedDataPipeConsumerHandle body) {
+    mojo::ScopedDataPipeConsumerHandle body,
+    absl::optional<mojo_base::BigBuffer> cached_metadata) {
```
Comment (bridiver) [inline components/body_sniffer/body_sniffer_url_loader.cc:70]: was this used at all in upstream mime_sniffer_url_loader? cc @ShivanKaul

File: browser/ui/views/brave_actions/brave_actions_container.cc [hunk 19]
```diff
@@ -534,3 +534,7 @@ void BraveActionsContainer::ToggleExtensionsMenu() {}
 bool BraveActionsContainer::HasAnyExtensions() const {
   return false;
 }
+
+void BraveActionsContainer::UpdateToolbarActionHoverCard(
+    ToolbarActionView* action_view,
+    ToolbarActionHoverCardUpdateType update_type) {}
```
Comment (bridiver) [inline browser/ui/views/brave_actions/brave_actions_container.cc:540]: we don't need to implement anything here? cc @petemill

File: ios/browser/providers/BUILD.gn [hunk 20]
```diff
@@ -35,6 +35,7 @@ group("brave_providers") {
     "//ios/chrome/browser/providers/text_zoom:chromium_text_zoom",
     "//ios/chrome/browser/providers/ui_utils:chromium_ui_utils",
     "//ios/chrome/browser/providers/url_rewriters:chromium_url_rewriters",
+    "//ios/chrome/browser/providers/user_feedback:chromium_user_feedback",
```
Comment (bridiver) [inline ios/browser/providers/BUILD.gn:38]: do we need this actual dep or can we provide a dummy provider for this? cc @kylehickinson

File: patches/ui-webui-resources-cr_elements-cr_shared_vars.css.patch [hunk 21]
```diff
@@ -0,0 +1,12 @@
+diff --git a/ui/webui/resources/cr_elements/cr_shared_vars.css b/ui/webui/resources/cr_elements/cr_shared_vars.css
+index 8c3e963e6368a3879c312021b358718d2470e1fb..8d6a71235ea2ac9f967fe840dd259e0b39ab45fa 100644
+--- a/ui/webui/resources/cr_elements/cr_shared_vars.css
++++ b/ui/webui/resources/cr_elements/cr_shared_vars.css
+@@ -5,6 +5,7 @@
+ /* #css_wrapper_metadata_start
+  * #type=vars
+  * #import=//resources/polymer/v3_0/paper-styles/color.js
++ * #import=//brave-resources/br_elements/shared_vars_css.m.js
```
Comment (bridiver) [inline patches/ui-webui-resources-cr_elements-cr_shared_vars.css.patch]: why is a new patch needed?

File: ios/browser/providers/BUILD.gn (hunk 20 above)
Comment (kylehickinson) [reply ios/browser/providers/BUILD.gn:38]: AFAIK we don't provide dummy providers for any of them, this file was only around to get rid of the MaterialComponents `+bundle` dep?

File: patches/ui-webui-resources-cr_elements-cr_shared_vars.css.patch (hunk 21 above)
Comment (petemill) [reply patches/ui-webui-resources-cr_elements-cr_shared_vars.css.patch]: It replaces the patch currently in master at https://github.com/brave/brave-core/blob/f5406a0a39929a6a5b524f950540b8fcfe657c84/patches/ui-webui-resources-cr_elements-shared_vars_css.html.patch

File: browser/brave_shields/brave_shields_web_contents_observer.cc [hunk 22]
```diff
@@ -214,22 +214,16 @@ void BraveShieldsWebContentsObserver::ReadyToCommitNavigation(
...
-              }
-            }
-          },
-          base::Unretained(this)));
+      [this](content::RenderFrameHost* rfh) {
+        this->GetBraveShieldsRemote(rfh)->SetAllowScriptsFromOriginsOnce(
+            this->allowed_script_origins_);
+        if (content::BrowserContext* context = rfh->GetBrowserContext()) {
+          if (PrefService* pref_service = user_prefs::UserPrefs::Get(context)) {
+            this->GetBraveShieldsRemote(rfh)->SetReduceLanguageEnabled(
```
[the remark this answers is not in the document]
Comment (emerick) [reply browser/brave_shields/brave_shields_web_contents_observer.cc]: Fixed here e89644d5b871a87d553f901a6c3db43800763fbe.

File: chromium_src/components/password_manager/core/common/password_manager_features.cc [hunk 23]
```diff
@@ -10,10 +10,8 @@
 namespace password_manager {
 namespace features {
 
-OVERRIDE_FEATURE_DEFAULT_STATES({{
-    {kPasswordImport, base::FEATURE_ENABLED_BY_DEFAULT},
-    {kUnifiedPasswordManagerDesktop, base::FEATURE_DISABLED_BY_DEFAULT},
-}});
+OVERRIDE_FEATURE_DEFAULT_STATES({{{kPasswordImport,
+                                   base::FEATURE_ENABLED_BY_DEFAULT}}});
```
[the remark this answers is not in the document]
Comment (emerick) [reply chromium_src/components/password_manager/core/common/password_manager_features.cc]: Fixed here 28c5e78b829c97868c1b1924711b0c3cc0f79ac7.

File: components/content_settings/core/browser/brave_content_settings_pref_provider.cc [hunk 24]
```diff
@@ -247,8 +247,8 @@ void BravePrefProvider::MigrateShieldsSettingsFromResourceIds() {
       base::Time last_modified =
           base::ValueToTime(settings_dict->Find(kLastModifiedPath))
               .value_or(base::Time());
-      for (const auto [key, value] : *resource_dict) {
-        const std::string& resource_identifier(key);
+      for (const auto [resource_key, resource_value] : *resource_dict) {
```
[the remark this answers is not in the document]
Comment (emerick) [reply components/content_settings/core/browser/brave_content_settings_pref_provider.cc]: Fixed here f1a388f84d397de448b8a6b946e3b99c30af1e48.

**Review by samartnik — APPROVED** [review]
Java++

**Review by darkdh — APPROVED** [review]
wallet and sync ++

File: chromium_src/content/public/browser/url_data_source.cc [hunk 25]
```diff
@@ -18,7 +18,7 @@ std::string URLDataSource::GetContentSecurityPolicy(
     case network::mojom::CSPDirectiveName::ScriptSrc:
       // Note: Do not add 'unsafe-eval' here. Instead override CSP for the
       // specific pages that need it, see context http://crbug.com/525224.
-      return "script-src chrome://resources chrome://brave-resources 'self';";
+      return "script-src chrome://resources 'self';";
```
Comment (bridiver) [inline chromium_src/content/public/browser/url_data_source.cc:21]: I don't think this is needed anymore and is actually probably unsafe because it doesn't call IsChromeUntrustedDataSource
The entire purpose of this was to add `chrome://brave-resources` so we should remove it

File: ios/browser/providers/BUILD.gn (hunk 20 above)
Comment (bridiver) [reply ios/browser/providers/BUILD.gn:38]: not sure, but wondering if we can reduce the size of the output by replacing some of these. I guess follow up either way

File: browser/brave_shields/brave_shields_web_contents_observer.cc [hunk 26]
```diff
@@ -214,22 +214,16 @@ void BraveShieldsWebContentsObserver::ReadyToCommitNavigation(
   }
 
   navigation_handle->GetWebContents()->ForEachRenderFrameHost(
```
Comment (bridiver) [inline browser/brave_shields/brave_shields_web_contents_observer.cc:216]: I wonder if this should be `ForEachRenderFrameHostIncludingSpeculative`

File: chromium_src/content/public/browser/url_data_source.cc (hunk 25 above)
Comment (cdesouza-chromium) [reply chromium_src/content/public/browser/url_data_source.cc]: Fixed in dcb57714ebeeb1c7854d3f99b3290f0c5367bd99.

**Review by goodov — APPROVED** [review]

File: browser/ui/views/brave_actions/brave_actions_container.cc (hunk 19 above)
Comment (petemill) [reply browser/ui/views/brave_actions/brave_actions_container.cc:540]: I don't think so. This is a new feature behind a flag and we haven't talked about what to do with it for Shields or Rewards yet. https://chromium.googlesource.com/chromium/src/+/c313724030f1fa5dcf1e2ca1ce127b00e1497aa2

**Review by petemill — APPROVED** [review]
Approving on behalf of js-deps-reviewers since there're no js deps changes.

**Review by goodov — APPROVED** [review]
NIK -> NAK conversion lgtm.

**Comment by bsclifton** [comment]
Going to go ahead and admin merge 😄 All approvals are in place except for @iefremov who is on vacation (and has given thumbs up on merge while he's out)
